src/ui_parts/global_menu.gd

```diff
@@ -0,0 +1,217 @@
+extends Node
+
+
+var global_rid: RID
+var appl_rid: RID
+var help_rid: RID
+
+var file_rid: RID
+var file_index: int
+var file_clear_svg_index: int
+var file_optimize_index: int
+var file_clear_assoc_index: int
+var file_reset_svg_index: int
+
+var edit_rid: RID
+var edit_index: int
+var tool_rid: RID
+var tool_index: int
+
+var view_rid: RID
+var view_index: int
+var view_show_grid_index: int
+var view_show_handles_index: int
+var view_rasterized_svg_index: int
+
+var snap_rid: RID
+var snap_index: int
+var snap_enable_index: int
+var snap_0125_index: int
+var snap_025_index: int
+var snap_05_index: int
+var snap_1_index: int
+var snap_2_index: int
+var snap_4_index: int
+
+
+func _enter_tree() -> void:
+	if not DisplayServer.has_feature(DisplayServer.FEATURE_GLOBAL_MENU):
+		queue_free()
+		return
+	# Included menus.
+	global_rid = NativeMenu.get_system_menu(NativeMenu.MAIN_MENU_ID)
+	appl_rid = NativeMenu.get_system_menu(NativeMenu.APPLICATION_MENU_ID)
+	help_rid = NativeMenu.get_system_menu(NativeMenu.HELP_MENU_ID)
+	# Custom menus.
+	_generate_main_menus()
+	_setup_menu_items()
+	GlobalSettings.keybinds_changed.connect(_reset_menu_items)
+	SVG.text_changed.connect(_on_svg_text_changed)
+
+
+func _notification(what: int) -> void:
+	if what == Utils.CustomNotification.LANGUAGE_CHANGED:
+		_clear_menu_items()
+		NativeMenu.remove_item(global_rid, snap_index)
+		NativeMenu.remove_item(global_rid, view_index)
+		NativeMenu.remove_item(global_rid, tool_index)
+		NativeMenu.remove_item(global_rid, edit_index)
+		NativeMenu.remove_item(global_rid, file_index)
+		NativeMenu.free_menu(file_rid)
+		NativeMenu.free_menu(edit_rid)
+		NativeMenu.free_menu(tool_rid)
+		NativeMenu.free_menu(view_rid)
+		NativeMenu.free_menu(snap_rid)
+		_generate_main_menus()
+		_setup_menu_items()
+
+
+func _generate_main_menus() -> void:
+	file_rid = NativeMenu.create_menu()
+	edit_rid = NativeMenu.create_menu()
+	tool_rid = NativeMenu.create_menu()
+	view_rid = NativeMenu.create_menu()
+	snap_rid = NativeMenu.create_menu()
+	file_index = NativeMenu.add_submenu_item(global_rid, TranslationServer.translate("File"), file_rid)
+	edit_index = NativeMenu.add_submenu_item(global_rid, TranslationServer.translate("Edit"), edit_rid)
+	tool_index = NativeMenu.add_submenu_item(global_rid, TranslationServer.translate("Tool"), tool_rid)
+	view_index = NativeMenu.add_submenu_item(global_rid, TranslationServer.translate("View"), view_rid)
+	snap_index = NativeMenu.add_submenu_item(global_rid, TranslationServer.translate("Snap"), snap_rid)
+
+
+func _reset_menu_items() -> void:
+	_setup_menu_items()
+
+
+func _clear_menu_items() -> void:
+	NativeMenu.clear(appl_rid)
+	NativeMenu.clear(help_rid)
+	NativeMenu.clear(file_rid)
+	NativeMenu.clear(edit_rid)
+	NativeMenu.clear(tool_rid)
+	NativeMenu.clear(view_rid)
+	NativeMenu.clear(snap_rid)
+
+
+func _setup_menu_items() -> void:
+	# Included App and Help menus.
```

Comment:
TranslationUtils should be used here to get the translated string of "open_settings". Same for the lines of code below. That way it's ensured that they are the same everywhere.